packages/components/inputs/time-input/src/time-input.spec.js

```diff
@@ -108,7 +109,7 @@ describe('TimeInput', () => {
     const event = { target: { value: '9' } };
     fireEvent.change(screen.getByLabelText('Input'), event);
 
-    await screen.findByText('Time: 9:00 AM');
+    await screen.findByText('Time: 9');
```

Comment:
TL;DR: change events should not format the time, otherwise the input becomes unusable. Only `onBlur` does the formatting.

Reply:
Perhaps I'm missing something but does this mean that users will see unformatted values while changing them until they move away the focus from the input field?

Reply:
Yes when users interact with the input they can potentially type whatever they want. This is important as otherwise the value keeps changing and the input becomes unusable.

You can see the issue in storybook: https://uikit.commercetools.com/?path=/story/components-inputs--timeinput

Reply:
Yip, I think the (if I follow along) UX of an input formatting as you type is subjectively horrible. At least I go bananas when I encounter them. You type and it formats. Maintaining cursor position in an input which formats while you type at the right position is super shifty.